app/(routes)/CreateScales/page.tsx

```diff
@@ -0,0 +1,186 @@
+        scaleDataMatrix,
+        staves,
+      }),
+    [rendererRef, setStaves, scaleDataMatrix, staves]
+  );
+
+  useEffect(() => {
+    setNotesAndCoordinates(() =>
+      generateYMinAndYMaxForAllNotes(147, notesArray)
+    );
+  }, []);
+
+  useEffect(() => {
+    initializeRenderer(rendererRef, container);
+    renderStavesAndNotes();
+  }, []);
+
+  useEffect(() => {
+    renderStavesAndNotes();
+  }, [scaleDataMatrix]);
+
+  useEffect(() => {
+    console.log("scaleDataMatrix Updated:", scaleDataMatrix);
+  }, [scaleDataMatrix]); // Listening to changes in scaleDataMatrix
+
+  const handleClick = (e: React.MouseEvent) => {
+    const { userClickY, userClickX } = getUserClickInfo(
+      e,
+      container,
+      staves[0]
+    );
+
+    let foundNoteData = notesAndCoordinates.find(
+      ({ yCoordinateMin, yCoordinateMax }) =>
+        userClickY >= yCoordinateMin && userClickY <= yCoordinateMax
+    );
+
+    console.log("foundNoteData with added original note", foundNoteData);
+
+    if (foundNoteData)
+      foundNoteData = {
+        ...foundNoteData,
+        userClickY: userClickY,
+      };
+
+    const barIndex = findBarIndex(staves, userClickX);
+
+    if (!foundNoteData) {
+      noNoteFound();
+      return;
+    }
+
+    let scaleDataMatrixCopy = scaleDataMatrix.map((scaleData) => [
+      ...scaleData,
+    ]);
+
+    let notesAndCoordinatesCopy = [...notesAndCoordinates];
+
+    const barOfScaleData = scaleDataMatrixCopy[barIndex].map(
+      (scaleData: ScaleData) => ({
+        ...scaleData,
+        staveNoteAbsoluteX: scaleData.staveNote
+          ? scaleData.staveNote.getAbsoluteX()
+          : 0,
+      })
+    );
+
+    const {
+      scaleDataMatrix: newScaleDataMatrix,
+      notesAndCoordinates: newNotesAndCoordinates,
+    } = HandleScaleInteraction(
+      foundNoteData,
+      tooManyBeatsInMeasure,
+      notesAndCoordinatesCopy,
+      "tooManyBeatsInMeasure",
+      barOfScaleData,
+      scaleDataMatrixCopy,
+      state,
+      userClickX,
+      userClickY,
+      barIndex
+    );
+    setNotesAndCoordinates(() => newNotesAndCoordinates);
+    setScaleDataMatrix(() => newScaleDataMatrix);
+  };
+
+  return (
+    <>
+      <div ref={container} onClick={handleClick} />
+      <CheckNumBeatsInMeasure
+        tooManyBeatsInMeasure={state.tooManyBeatsInMeasure}
+        openEnterNotes={dispatch}
+      />
+      <CheckIfNoteFound
+        noNoteFound={state.noNoteFound || false}
+        openEnterNotes={dispatch}
+      />
+      <div className="mt-2 ml-3">
+        {modifyStaveNotesButtonGroup.map((button) => {
+          return (
+            <BlueButton
+              key={button.text}
+              onClick={button.action}
+              isEnabled={button.isEnabled}
+            >
+              {button.text}
+            </BlueButton>
+          );
+        })}
+        <BlueButton onClick={eraseMeasures}>Erase Measure</BlueButton>
+      </div>
+    </>
+  );
+};
+
+export default ManageScales;
```

Comment:
Instead of ManageScales being a route at (routes)/CreateScales/page.tsx, it should be its own component (either called ManageScales or CreateScales) in the component folder. Then we can import it to all of the ScalesNotate pages in the exam.